outputs/hn_zones/plot_comparison_of_hn_zones.py

```diff
@@ -0,0 +1,387 @@
+    else:  # score == 0
+        subset = hp_suitability_gdf[hp_suitability_gdf["DESNZ_pilot_fraction"] == 0]
+        other = hp_suitability_gdf[hp_suitability_gdf["DESNZ_pilot_fraction"] > 0]
+
+    if subset.empty:
+        logging.warning(f"No regions found for the given score threshold: {score}")
+
+    subset.plot(column="absolute_error", ax=ax, legend=True, cmap="inferno")
+    other.plot(ax=ax, edgecolor="black", facecolor="none", linewidth=0.5)
+
+    # Update legend labeling for clarity
+    cbar = ax.get_figure().get_axes()[1]
+    cbar.set_ylabel(
+        "Absolute Error (|Avg. HN Score - DESNZ HN Score|)", rotation=90, labelpad=15
+    )
+
+    plt.title(title)
+    plt.axis("off")
+    plt.savefig(os.path.join(output_dir, f"{filename}.png"))
+    plt.savefig(os.path.join(output_dir, f"{filename}.pdf"))
+    plt.show()
+
+
+def plot_hn_avg_score_vs_fraction_threshold(
+    average_hn_scores_coverage_df: pl.DataFrame,
+    fraction_thresholds_col: str,
+    hn_avg_scores_col: str,
+    filename: str,
+    output_dir: str = OUTPUT_DIR,
+):
+    """
+    Plot the average Nesta HN score against DESNZ Pilot Fraction Threshold, fit a regression line,
+    and display R^2.
+
+    Args:
+        average_hn_scores_coverage_df (pl.DataFrame): DataFrame with average Nesta HN scores and DESNZ Pilot Fraction.
+        fraction_thresholds_col (str): Column name for DESNZ Pilot Fraction Threshold.
+        hn_avg_scores_col (str): Column name for HN_N Avg Score Weighted.
+    """
+    logging.info("Plotting average HN score vs DESNZ Pilot Fraction Threshold...")
+    fraction_thresholds = average_hn_scores_coverage_df[
+        fraction_thresholds_col
+    ].to_list()
+    hn_avg_scores = average_hn_scores_coverage_df[hn_avg_scores_col].to_list()
+
+    x = np.array(fraction_thresholds)
+    y = np.array(hn_avg_scores)
+
+    # Fit a linear model
+    coefficients = np.polyfit(x, y, 1)
+    model = np.poly1d(coefficients)
+    predicted_y = model(x)
+    r_squared = r2_score(y, predicted_y)
+
+    y_min = min(hn_avg_scores) - 0.05
+    y_max = max(hn_avg_scores) + 0.05
+
+    fig, ax = plt.subplots(figsize=(10, 6))
+    ax.plot(
+        fraction_thresholds,
+        hn_avg_scores,
+        marker="o",
+        linestyle="-",
+        linewidth=2,
+        label="Data",
+    )
+    ax.plot(
+        fraction_thresholds,
+        predicted_y,
+        color="red",
+        linestyle="--",
+        linewidth=2,
+        label="Regression line",
+    )
+
+    ax.set_ylim(y_min, y_max)
+    ax.spines["top"].set_visible(False)
+    ax.spines["right"].set_visible(False)
+
+    # Break mark on the y-axis
+    d = 0.015
+    kwargs = dict(transform=ax.transAxes, color="k", clip_on=False)
+    ax.plot((-d, +d), (-d, +d), **kwargs)
+    ax.plot((-d, +d), (1 - d, 1 + d), **kwargs)
+
+    ax.set_title(
+        "Average Nesta HN Score vs Fraction of DESNZ HN zone in LSOA", fontsize=14
```

Comment:
```suggestion
        "Average Nesta HN Score vs Fraction of LSOA in DESNZ Heat Network zone", fontsize=14
```
Should it be this way round? Otherwise lmk because I got confused!

Reply:
It's the fraction of the DESNZ HN zone in the LSOA. We are increasing the threshold, so as we increase the amount of DESNZ HN zone in the LSOA, our average Nesta HN score. This is what we expect intuitively!

Reply:
thanks for explaining, that makes sense :D